src/core/meshcop/tcat_agent.hpp

```diff
@@ -0,0 +1,382 @@
+
+private:
+    Error HandleGeneral(ot::Message &aIncommingMessage,
+                        ot::Message &aOutgoingMessage,
+                        uint8_t      aTlvType,
+                        uint32_t     aOffset,
+                        uint32_t     aLength,
+                        bool        &aHasResponse);
+    Error HandleCommissioning(ot::Message &aIncommingMessage,
+                              ot::Message &aOutgoingMessage,
+                              uint8_t      aTlvType,
+                              uint32_t     aOffset,
+                              uint32_t     aLength,
+                              bool        &aHasResponse);
+    Error HandleExtraction(ot::Message &aIncommingMessage,
+                           ot::Message &aOutgoingMessage,
+                           uint8_t      aTlvType,
+                           uint32_t     aOffset,
+                           uint32_t     aLength,
+                           bool        &aHasResponse);
+    Error HandleDecommissioning(ot::Message &aIncommingMessage,
+                                ot::Message &aOutgoingMessage,
+                                uint8_t      aTlvType,
+                                uint32_t     aOffset,
+                                uint32_t     aLength,
+                                bool        &aHasResponse);
+    Error HandleApplication(ot::Message &aIncommingMessage,
+                            ot::Message &aOutgoingMessage,
+                            uint8_t      aTlvType,
+                            uint32_t     aOffset,
+                            uint32_t     aLength,
+                            bool        &aHasResponse);
+    Error HandleCCM(ot::Message &aIncommingMessage,
+                    ot::Message &aOutgoingMessage,
+                    uint8_t      aTlvType,
+                    uint32_t     aOffset,
+                    uint32_t     aLength,
+                    bool        &aHasResponse);
+
+    Error HandleCommand(CommandType aCommand, ot::Message &aOutgoingMessage);
+    Error HandleActiveDataset(ot::Message &aIncommingMessage, uint32_t aOffset, uint32_t aLength);
+
+    static constexpr uint16_t kJoinerUdpPort = OPENTHREAD_CONFIG_JOINER_UDP_PORT;
+
+    JoinerPskd                       mJoinerPskd;
+    VendorInfo                      *mVendorInfo;
+    Callback<JoinCallback>           mJoinCallback;
+    Callback<AppDataReceiveCallback> mAppDataReceiveCallback;
+    CertificateAuthorizationField    mCommissionerAuthorizationField;
+    CertificateAuthorizationField    mDeviceAuthorizationField;
+    TcatMessageType                  mCurrentApplicationLayer;
+    char                             mCurrentServiceName[OT_TCAT_MAX_SERVICE_NAME_LENGTH + 1];
+    bool                             mEnabled : 1;
+    bool                             mConnected : 1;
+    bool                             mAlreadyCommissioned : 1;
```

Comment:
We can consider defining `State` `enum` here instead of the set of `bool`?

The good thing about `State` is we avoid combinations that are invalid (e.g., I guess we cannot have `mEnabled == false` with `mConnected == true`?)
```c++
enum State : uint8_t 
{
     kStateDisabled,
     kStateEnabled,
     kStateConnected,
     ...
};
```

Reply:
OK, will do for mEnabled & mConnected.
IsEnabled() still writes nicely as (state !=kStateDisabled).

mAlreadyCommissioned is not a current state. It is to remember if the session was started in commissioned state. This needs to stay bool I believe.

Reply:
Implemented as suggested.